### PR TITLE
Only reindex container metadata on Foreman server

### PR DESCRIPTION
**Pull request by sjha4**
We need to confine katello db indexing to run only on Satellite and not capsules. Refactoring a little to do that.

### CODE REVIEW

File: definitions/procedures/repositories/index_katello_repositories_container_metadata.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Procedures::Repositories
+  class IndexKatelloRepositoriesContainerMetatdata < ForemanMaintain::Procedure
+    metadata do
+      description 'Import container manifest metadata into Katello'
+      confine do
+        feature(:satellite)
```
Comment (ehelms) [inline definitions/procedures/repositories/index_katello_repositories_container_metadata.rb:6]: Is this procedure only applicable to Satellite? The description and text below both use 'Katello'. But if this is something a user of Katello can/should also be having run on a Foreman then we need to update the confines.

File: definitions/procedures/repositories/index_katello_repositories_container_metadata.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module Procedures::Repositories
+  class IndexKatelloRepositoriesContainerMetatdata < ForemanMaintain::Procedure
+    metadata do
+      description 'Import container manifest metadata into Katello'
```
Comment (ehelms) [inline definitions/procedures/repositories/index_katello_repositories_container_metadata.rb:4]:
```suggestion
      description 'Import container manifest metadata'
```

File: definitions/procedures/repositories/index_katello_repositories_container_metadata.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      confine do
+        feature(:katello)
+      end
+    end
+
+    def run
+      with_spinner(('Adding image metadata to katello. You can continue using the ' \
```
Comment (ehelms) [inline definitions/procedures/repositories/index_katello_repositories_container_metadata.rb:11]:
```suggestion
      with_spinner(('Adding image metadata. You can continue using the ' \
```

**Comment by ehelms** [comment]
You might consider commit message like 'Only reindex container metadata on Foreman server'

**Comment by sjha4** [comment]
@ehelms : Thanks..Updated PR and commit message.. ❤️

**Review by ehelms — APPROVED** [review]

**Pull request merged**